mbf_abstract_nav/src/abstract_controller_execution.cpp

```diff
@@ -185,6 +186,63 @@ void AbstractControllerExecution::setVelocityCmd(const geometry_msgs::TwistStamp
   // TODO so there should be no loss of information in the feedback stream
 }
 
+bool AbstractControllerExecution::checkCmdVelIgnored(const geometry_msgs::Twist& cmd_vel)
+{
+  // check if the velocity ignore check is enabled or not
+  if (cmd_vel_ignored_tolerance_ <= 0.0)
+  { 
+    // set to zero if not already zero
+    if (!first_ignored_time_.is_zero())
+    {
+      first_ignored_time_ = ros::Time();
+    }
+    return false;
+  }
+
+  const bool robot_stopped = robot_info_.isRobotStopped(1e-3, 1e-3);
+
+  // compute linear and angular velocity magnitude
+  const double cmd_linear = std::hypot(cmd_vel.linear.x, cmd_vel.linear.y);
+  const double cmd_angular = std::abs(cmd_vel.angular.z);
+
+  const bool cmd_is_zero = cmd_linear < 1e-3 && cmd_angular < 1e-3;
+
+  // velocity is not being ignored
+  if (!robot_stopped || cmd_is_zero)
+  {
+    // set to zero if not already zero
```

Comment:
F: this comment says the same as the code.  Write comments only to clarify non-obvious steps

Reply:
same goes for the next 2 comments